project_scrum/models/project_sprint.py

```diff
@@ -0,0 +1,135 @@
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+from odoo.tools import LazyTranslate
+
+_lt = LazyTranslate(__name__, default_lang="en_US")
+
+DATE_OPTIONS = [
+    ("1_weeks", str(_lt("1 Week"))),
+    ("2_weeks", str(_lt("2 Weeks"))),
+    ("1_months", str(_lt("1 Month"))),
+    ("2_months", str(_lt("2 Month"))),
+    ("1_years", str(_lt("1 Year"))),
+    ("2_years", str(_lt("2 Years"))),
+    ("custom", str(_lt("Custom"))),
+]
+
+
+class ProjectSprint(models.Model):
+    _name = "project.sprint"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+    _description = "Project Sprint"
+    _sql_constraints = [
+        (
+            "date_check",
+            "CHECK (date_start <= date_end)",
+            "Error: End date must be greater than start date!",
+        ),
+    ]
+
+    name = fields.Char(required=True, tracking=True)
+    user_ids = fields.Many2many(
+        comodel_name="res.users",
+        string="Members",
+        required=True,
+        domain="[('share', '=', False), ('active', '=', True)]",
+        tracking=True,
+        relation="project_sprint_user_rel",
+    )
+    description = fields.Text(tracking=True)
+    project_id = fields.Many2one(
+        comodel_name="project.project",
+        string="Project",
+        tracking=True,
+    )
+    task_ids = fields.One2many(
+        comodel_name="project.task",
+        inverse_name="sprint_id",
+        string="Tasks",
+        domain="[('project_id', '=', project_id)]",
+    )
+    date_start = fields.Date(
+        string="Start Date", default=fields.Date.today, required=True
+    )
+    date_option = fields.Selection(
+        selection=DATE_OPTIONS, default=DATE_OPTIONS[0][0], required=True
+    )
+    date_end = fields.Date(
+        string="End Date",
+        required=True,
+        compute="_compute_date_end",
+        store=True,
+        readonly=False,
+    )
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("in_progress", "In progress"),
+            ("done", "Done"),
+        ],
+        default="draft",
+    )
+    tasks_count = fields.Integer(compute="_compute_tasks_count")
+
+    def _compute_tasks_count(self):
+        for sprint in self:
+            sprint.tasks_count = len(sprint.task_ids)
```

Comment:
I think this code can be improved by using `_read_group`; it's more efficient. What do you think?

Reply:
@carlos-lopez-tecnativa , one2many fields are not supported in read_group I think. Moreover, it wont make much difference I think

Reply:
Please see examples in project
https://github.com/odoo/odoo/blob/1939d570616ad1ae7b6f182a76fc9f8f0b018fe1/addons/project/models/project_project.py#L43-L47

```suggestion
    def _compute_tasks_count(self):
        tasks_count_by_sprint = dict(self.env['project.task']._read_group([('sprint_id', 'in', self.ids), ['sprint_id'], ['__count']))
        for sprint in self:
            sprint.tasks_count = tasks_count_by_sprint.get(sprint, 0)
```